File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def largestValues(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        ans = []
        def dfs(lvl, node):
            if not node:
                return 

            if lvl == len(ans):
                ans.append(node.val)
            
            ans[lvl] = max(node.val, ans[lvl])

            dfs(lvl+1, node.left)
            dfs(lvl+1, node.right)

        dfs(0, root)

        return ans

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public void dfs(int lvl, TreeNode root, List<Integer> res) {
        if (root == null)
            return;
        if (lvl == res.size())
            res.add(root.val);
        else
            res.set(lvl, Math.max(root.val, res.get(lvl)));
        dfs(lvl + 1, root.left, res);
        dfs(lvl + 1, root.right, res);
        return;
    }

    public List<Integer> largestValues(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        dfs(0, root, res);
        return res;
    }
}